### PR TITLE
Copy into integer cube map texture should work

### PR DESCRIPTION
**Pull request by qkmiao**
This is a regression test for crbug.com/712117.

### CODE REVIEW

File: sdk/tests/conformance2/textures/misc/00_test_list.txt [hunk 1]
```diff
@@ -2,6 +2,7 @@
 --min-version 2.0.1 angle-stuck-depth-textures.html
 canvas-remains-unchanged-after-used-in-webgl-texture.html
 --min-version 2.0.1 compressed-tex-from-pbo-crash.html
+copy-texture-cube-map-AMD-bug.html
```
Comment (zhenyao) [inline sdk/tests/conformance2/textures/misc/00_test_list.txt]: Can you add --min-version here?

File: sdk/tests/conformance2/textures/misc/copy-texture-cube-map-AMD-bug.html [hunk 2]
```diff
@@ -0,0 +1,92 @@
...
+<script>
+"use strict";
+var wtu = WebGLTestUtils;
+var gl;
+
+function runTest() {
+  var width = 16;
+  var height = 16;
+  var level = 0;
+  var internalformat = gl.RGBA8UI;
```
Comment (zhenyao) [inline sdk/tests/conformance2/textures/misc/copy-texture-cube-map-AMD-bug.html]: Can we expand the test to cover also RGBA8I, RGBA8, RGBA32F?

Comment (qkmiao) [inline sdk/tests/conformance2/textures/misc/copy-texture-cube-map-AMD-bug.html]: I think RGBA32F isn't supported by CopyTexImage2D in chrome now. Should we support floating formats  if https://www.khronos.org/registry/webgl/extensions/EXT_color_buffer_float/ is enabled? I didn't see explicit specification about this in the extension spec.

File: sdk/tests/conformance2/textures/misc/00_test_list.txt (hunk 1 above)
Comment (qkmiao) [inline sdk/tests/conformance2/textures/misc/00_test_list.txt]: I added min-version 2.0.1 for this test.

File: sdk/tests/conformance2/textures/misc/copy-texture-cube-map-AMD-bug.html (hunk 2 above)
Comment (qkmiao) [inline sdk/tests/conformance2/textures/misc/copy-texture-cube-map-AMD-bug.html]: I added other formats. We can easily add floating formats in the test if we decide to.

**Comment by qkmiao** [comment]
Thanks @zhenyao . I updated this pr to fix your comments. PTAL.

**Comment by msc-** [comment]
> Should we support floating formats if https://www.khronos.org/registry/webgl/extensions/EXT_color_buffer_float/ is enabled? I didn't see explicit specification about this in the extension spec.

That's because it is mostly covered in the core spec. The extension spec. modifies the CopyTexImage INVALID_OPERATION error on p.139 so it is only raised when trying to copy floating point data to a format that is not floating- or fixed-point.

ReadPixels is modified to support RGBA and FLOAT. (Table 3.15 should have been modified to include RGBA and FLOAT but I the modification to ReadPixels makes this combination valid.

Table 3.17 is modified to include RG{B,BA}32F.

The core spec. says

> If internalformat is sized, the internal format of the new texel array is internalformat, and this is also the new texel array’s effective internal format. If the component sizes of internalformat do not exactly match the corresponding component sizes of the source buffer’s effective internal format, described below, an INVALID_OPERATION error is generated.

So the conclusion is that copying RGBA32F to RGBA32F is supported.

**Comment by qkmiao** [comment]
@msc- I think you are right. Thanks. I will fix this issue in chrome and add conformance test to check this.

**Comment by zhenyao** [comment]
LGTM
Please add float test case with extension check in a followup CL

**Pull request merged**